File: fabric/src/main/java/juuxel/adorn/compat/PromenadeCompat.java
package juuxel.adorn.compat;

import juuxel.adorn.block.variant.BlockVariant;
import juuxel.adorn.block.variant.PrefixedBlockVariantSet;

import java.util.List;

public final class PromenadeCompat extends PrefixedBlockVariantSet {
    @Override
    public String getModId() {
        return "promenade";
    }

    @Override
    public List<BlockVariant> getWoodVariants() {
        return createVariants(
            BlockVariant.Wood::new,
            "dark_amaranth",
            "palm",
            "sakura"
        );
    }
}

File: fabric/src/main/kotlin/juuxel/adorn/compat/PromenadeCompat.kt
package juuxel.adorn.compat

import juuxel.adorn.block.variant.BlockVariant
import juuxel.adorn.block.variant.BlockVariantSet

object PromenadeCompat : BlockVariantSet {
    override val woodVariants = listOf(
        "dark_amaranth",
        "palm",
        "sakura",
    ).map { BlockVariant.Wood("promenade/$it") }
}
